### PR TITLE
Jsunit tests2

### PR DESCRIPTION
**Pull request by waxenegger**
Forced push killed https://github.com/ome/omero-iviewer/pull/27 ... so there we go again...

The console runs of the jsunit tests exhibited an error while the browser runs worked.

Phantomjs, a headless browser test tool failed with a parse error while not only browser engines of chrome, google, IE but also node, mocha, jslint and http://esprima.org/ took no issue with the debug version of the file in question. Since an error on that level is a nightmare to get better info on in phantomjs it took an excessive amount of time to determine that something in the whitespacing or js docs / comments was the problem. 

As a preventive measure against that the ol3 debug/test js is now stripped of all comments and whitespace (but not compressed/mangled) before using it for the tests.

Also, the location of the test got moved into the root of the project to use it for future tests that concern the entire project and not just the ol3 bits.

As part of making for an easier analysis of such errors in the future a few debug files for mocha, phantomjs and the browser were added.

Here is what the console output should look like, in the success and error case which makes ant fail with an exit value non 0 so that the tests could now be integrated with other tools.

screen_tests

screen_test_failure

TEST:  Clone the project locally, cd to the root, then execute `ant unit-tests`

### CODE REVIEW

File: test/unit/debug_mocha.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/**
+ * THIS IS NOT USED FOR THE UNIT TESTS OTHER THAN FOR DEBUGGING ON DEMAND!!!
+ * THE TESTS LISTED IN THIS FILE DO NOT NEED TO BE KEPT UP TO DATE !!!!
+ *
+ * In particular, you can debug mocha this way:
+ * mocha debug_mocha.js
+ *
+ * IMPORTANT:
+ * mocha is not agnostic of browser enviroments, hence its use with
```
Comment (jburel) [inline test/unit/debug_mocha.js]: typo ``environments``

File: test/unit/debug_tests.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ *                     not_loaded--;
+ *                     if (not_loaded === 0) mocha.run();
+ *                 })
+ *             });
+ *         }, 2000);
+ *
+ *  2. uncomment the line in the head tag:
+ * <!--script type="text/javascript" src="debug_tests.js"></script-->
+ */
+function addScript( src,callback) {
```
Comment (jburel) [inline test/unit/debug_tests.js]: extra space before ``src`` and missing after ``,``

File: test/unit/debug_tests.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  var s = document.createElement( 'script' );
```
Comment (jburel) [inline test/unit/debug_tests.js]: extra space not consistent

File: test/unit/debug_tests.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  s.setAttribute( 'src', src );
```
Comment (jburel) [inline test/unit/debug_tests.js]: same

File: test/unit/debug_tests.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  s.onload=callback;
+  s.onerror=function(what) {
+      console.error(what);
+  };
+  document.head.appendChild( s );
```
Comment (jburel) [inline test/unit/debug_tests.js]: same

**Comment by jburel** [comment]
``
Getting rid of js docs and comments (for phantomjs) ....
     [echo] Running Test Suite ....
     [exec] 
     [exec] 
     [exec]   Conversion
     [exec]     ✓ convertRgbaColorFormatToObject
     [exec]     ✓ convertHexColorFormatToObject
     [exec]     ✓ convertColorObjectToHex
     [exec]     ✓ convertColorObjectToRgba
     [exec]     ✓ convertColorToSignedInteger
     [exec]     ✓ pointToJsonObject
     [exec]     ✓ ellipseToJsonObject
     [exec]     ✓ rectangleToJsonObject
     [exec]     ✓ lineToJsonObject
     [exec]     ✓ polylineToJsonObject
     [exec]     ✓ labelToJsonObject
     [exec]     ✓ polygonToJsonObject
     [exec]     ✓ integrateStyleAndMiscIntoJsonObject1
     [exec]     ✓ integrateStyleAndMiscIntoJsonObject2
     [exec]     ✓ toJsonObject
     [exec] 
     [exec]   Misc
     [exec]     ✓ prepareResolutions
     [exec]     ✓ parseSvgStringForPolyShapes
     [exec] 
     [exec]   Net
     [exec]     ✓ checkAndSanitizeServerAddress
     [exec]     ✓ checkAndSanitizeUri
     [exec] 
     [exec]   Regions
     [exec]     ✓ featureFactory
     [exec]     ✓ generateRegions
     [exec] 
     [exec]   Style
     [exec]     ✓ createFeatureStyle
     [exec] 
     [exec]   Geometries
     [exec]     ✓ createEllipse
     [exec]     ✓ createRectangle
     [exec]     ✓ createLabel
     [exec]     ✓ createPolyline
     [exec] 
     [exec] 
     [exec]   26 passing (21ms)
     [exec] 
     [exec] 

BUILD SUCCESSFUL
``

**Comment by jburel** [comment]
Thanks for updating the readme
Do you want to change the extension to ``.rst``? in this PR

**Comment by waxenegger** [comment]
sure, seems too tiny a change to justify a new PR.

**Comment by jburel** [comment]
you could also use the following formatting for hyperlink
e.g. [Phantom](url)

**Comment by jburel** [comment]
Thanks
Much nicer
merging

**Pull request merged**